public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -262,7 +262,41 @@ public ConceptAnalysis apply(ConceptAnalysis ca) {
                 conceptService.searchConcepts(searchConceptsRequest.getQuery(), convertedConceptFilter,
                         searchConceptsRequest.getVocabularyIds(), domainIds, maxResults);
         ConceptListResponse response = new ConceptListResponse();
-        response.setItems(concepts.getContent().stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+        List<Concept> matchedConcepts = concepts.getContent();
+        List<Concept> conceptCodeMatches = new ArrayList<>();
+        for(Concept con : matchedConcepts){
+            String conceptCode = con.getConceptCode();
+            Long conceptId = con.getConceptId();
+            if(!con.getStandardConcept().equals("S")){
+                if(conceptCode.equals(searchConceptsRequest.getQuery())) {
+
+                    conceptCodeMatches.add(con);
+
+                    response.setMatchType("ConceptCode");
+
+                    List<Concept> std_concepts = conceptDao.findStandardConcept(con.getConceptId());
+                    response.setStandardConcept(std_concepts.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+
+                    response.setItems(conceptCodeMatches.stream().map(TO_CLIENT_CONCEPT).collect(Collectors.toList()));
+
+                    return ResponseEntity.ok(response);
```

Comment:
(As mentioned previously, I wouldn't necessarily return just the non-standard concept here, but I guess this is a product question.)

Reply:
Changed the method to return the non standard ones along with the standard ones they are mapped to and any name concept matches too.